src/app/public/modules/phone-field/phone-field.component.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  OnDestroy,
+  OnInit,
+  Output,
+  Input
+} from '@angular/core';
+
+require('intl-tel-input/build/js/utils');
+
+require('intl-tel-input/build/js/intlTelInput');
+
+/**
+ * NOTE: We can not type these due the the current @types/intl-tel-input version having an
+ * undeclared type which causes linting errors.
+ */
+declare var intlTelInputUtils: any;
+declare var intlTelInputGlobals: any;
+
+import {
+  SkyPhoneFieldCountry
+} from './types';
+
+@Component({
+  selector: 'sky-phone-field',
+  templateUrl: './phone-field.component.html',
+  styleUrls: ['./phone-field.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyPhoneFieldComponent implements OnDestroy, OnInit {
+
+  @Input()
+  public set defaultCountry(value: string) {
+    if (value !== this._defaultCountry) {
+      this._defaultCountry = value;
+
+      this.defaultCountryData = this.countryData.find(country => country.iso2 === value);
+      this.sortCountriesWithSelectedAndDefault(this.selectedCountry);
+    }
+  }
+
+  public get defaultCountry(): string {
+    return this._defaultCountry;
+  }
+
+  @Output()
+  public selectedCountryChange = new EventEmitter<SkyPhoneFieldCountry>();
+
+  public countryData: SkyPhoneFieldCountry[];
```

Comment:
I'm being pedantic only because it's a `public` property, but could we name this `countries`, since the modifier "data" doesn't add much?

Reply:
Done